src/printer-htmlparser2.js

```diff
@@ -103,7 +111,15 @@ function printAttributes(attribs) {
         if (attribs[name] === "") {
           return name;
         }
-        return concat([name, '="', attribs[name], '"']);
+
+        // FastPath requires the next node to be a property of the current one
+        node._currentAttribute = {
```

Comment:
We're already doing it for css, I would also do it for HTML. We haven't done much performance work so far and it's been reasonable.

I'd rather do the right thing for now and when we decide that perf is important, then measure and optimize the things that are actually taking a lot of time and make trade-offs at that point.